core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -348,4 +365,75 @@ private static String buildErrorMessage(String stage, Throwable e) {
         }
         return errorMessage.toString();
     }
+    
+    static final String PLUGIN_LIB_PATTERN = "glob:**.{jar,zip}";
+    private static void setupPlugins(Environment environment) throws IOException {
+        ESLogger logger = Loggers.getLogger(Bootstrap.class);
+
+        Path pluginsDirectory = environment.pluginsFile();
+        if (!isAccessibleDirectory(pluginsDirectory, logger)) {
+            return;
+        }
+
+        ClassLoader classLoader = ClassLoader.getSystemClassLoader();
+        Class<?> classLoaderClass = classLoader.getClass();
+        Method addURL = null;
+        while (!classLoaderClass.equals(Object.class)) {
+            try {
+                addURL = classLoaderClass.getDeclaredMethod("addURL", URL.class);
+                addURL.setAccessible(true);
+                break;
+            } catch (NoSuchMethodException e) {
+                // no method, try the parent
+                classLoaderClass = classLoaderClass.getSuperclass();
+            }
+        }
+
+        if (addURL == null) {
```

Comment:
In the event that the ClassLoader doesn't contain an addURL method this will be tripped. I think this is more of a WARNING level or and ERROR than a debug. Right?